### ISSUE TITLE
Does the function 'solve_quadprog(G,g, ... ,x)' change the value of 'G'? 

### ISSUE THREAD

**Issue by 1kingofhand** [liuq/QuadProgpp#30]
I find there is a 'cholesky_decomposition(G)' option in the source code, which return 'void', thereby changing the value of 'G'.  Hence, there is a problem. Does the function 'solve_quadprog(G,g, ... ,x)' change the value of 'G'?  When solving a MPC, we need to use 'solve_quadprog(G,g, ... ,x)' in a loop, such a issue may be obmitted and causes mistakes in the results.

**Comment by liuq** [comment]
Yes, the original Matrix is not preserved since it is passed by reference (not as a const). This is common practice in C/C++. You can eventually make yourself a copy of the matrix before calling the function.All the best,Luca Di GasperoSent from iPhone: forgive brevity and mispellingsIl giorno 18 dic 2024, alle ore 07:40, 1kingofhand ***@***.***> ha scritto:﻿
I find there is a 'cholesky_decomposition(G)' option in the source code, which return 'void', thereby changing the value of 'G'.  Hence, there is a problem. Does the function 'solve_quadprog(G,g, ... ,x)' change the value of 'G'?  When solving a MPC, we need to use 'solve_quadprog(G,g, ... ,x)' in a loop, such a issue may be obmitted and causes mistakes in the results.

**Comment by 1kingofhand** [comment]
> Yes, the original Matrix is not preserved since it is passed by reference (not as a const). This is common practice in C/C++. You can eventually make yourself a copy of the matrix before calling the function.All the best,Luca Di GasperoSent from iPhone: forgive brevity and mispellingsIl giorno 18 dic 2024, alle ore 07:40, 1kingofhand ***@***.***> ha scritto:﻿ I find there is a 'cholesky_decomposition(G)' option in the source code, which return 'void', thereby changing the value of 'G'. Hence, there is a problem. Does the function 'solve_quadprog(G,g, ... ,x)' change the value of 'G'? When solving a MPC, we need to use 'solve_quadprog(G,g, ... ,x)' in a loop, such a issue may be obmitted and causes mistakes in the results. —Reply to this email directly, view it on GitHub, or unsubscribe.You are receiving this because you are subscribed to this thread.Message ID: ***@***.***>

Thanks  for replying! I got it.
